### PR TITLE
Fix hardhat & slither using remix desktop

### PR DESCRIPTION
**Pull request by yann300**
fix https://github.com/ethereum/remix-project/issues/1456
fix https://github.com/ethereum/remix-desktop/issues/67

### CODE REVIEW

File: libs/remixd/src/services/slitherClient.ts [hunk 1]
```diff
@@ -114,21 +114,30 @@ export class SlitherClient extends PluginClient {
...
+      }
+      if (solcArgs) {
+        solcArgs = `--solc-args "${solcArgs.trimStart()}"`
+      }
       const solcRemaps = remaps ? `--solc-remaps "${remaps}"` : ''
 
       const outputFile: string = 'remix-slitherReport_' + Math.floor(Date.now() / 1000) + '.json'
       const cmd: string = `slither ${filePath} ${solcArgs} ${solcRemaps} --json ${outputFile}`
       console.log('\x1b[32m%s\x1b[0m', '[Slither Analysis]: Running Slither...')
+      console.log(cmd)
```
Comment (Aniket-Engg) [inline libs/remixd/src/services/slitherClient.ts:140]: This can be removed

File: libs/remixd/src/services/slitherClient.ts [hunk 2]
```diff
@@ -140,6 +149,13 @@ export class SlitherClient extends PluginClient {
         if (existsSync(outputFileAbsPath)) {
           let report = readFileSync(outputFileAbsPath, 'utf8')
           report = JSON.parse(report)
+          try {
+            unlink(outputFileAbsPath, (err) => {
```
Comment (Aniket-Engg) [inline libs/remixd/src/services/slitherClient.ts:153]: We can also use `unlinkSync`

Comment (yann300) [reply libs/remixd/src/services/slitherClient.ts:153]: we don't really need to wait for the result.

Comment (Aniket-Engg) [reply libs/remixd/src/services/slitherClient.ts:152]: ok

File: libs/remixd/src/services/slitherClient.ts [hunk 3]
```diff
@@ -140,6 +148,13 @@ export class SlitherClient extends PluginClient {
...
```
Comment (Aniket-Engg) [inline libs/remixd/src/services/slitherClient.ts:152]: not sure if we use try-catch with callback

**Review by Aniket-Engg — APPROVED** [review]

**Review by bunsenstraat — APPROVED** [review]

File: libs/remixd/src/services/slitherClient.ts [hunk 4]
```diff
@@ -140,6 +148,9 @@ export class SlitherClient extends PluginClient {
         if (existsSync(outputFileAbsPath)) {
           let report = readFileSync(outputFileAbsPath, 'utf8')
           report = JSON.parse(report)
+          unlink(outputFileAbsPath, (err) => {
+            console.log(err)
```
Comment (bunsenstraat) [inline libs/remixd/src/services/slitherClient.ts:152]: if (err) console.log(err)
otherwise it always prints null everytime

**Review by bunsenstraat — CHANGES_REQUESTED** [review]

**Review by bunsenstraat — APPROVED** [review]

**Pull request merged**